diff --git a/node_modules/@google-cloud/storage/node_modules/teeny-request/build/src/index.js b/node_modules/@google-cloud/storage/node_modules/teeny-request/build/src/index.js
index af5d15e..8a9bf77 100644
--- a/node_modules/@google-cloud/storage/node_modules/teeny-request/build/src/index.js
+++ b/node_modules/@google-cloud/storage/node_modules/teeny-request/build/src/index.js
@@ -168,31 +168,34 @@ function teenyRequest(reqOpts, callback) {
         // Stream mode
         const requestStream = streamEvents(new stream_1.PassThrough());
         // eslint-disable-next-line @typescript-eslint/no-explicit-any
-        let responseStream;
-        requestStream.once('reading', () => {
-            if (responseStream) {
-                (0, stream_1.pipeline)(responseStream, requestStream, () => { });
-            }
-            else {
-                requestStream.once('response', () => {
-                    (0, stream_1.pipeline)(responseStream, requestStream, () => { });
-                });
-            }
-        });
+        // let responseStream;
+        // requestStream.once('reading', () => {
+        //    if (responseStream) {
+        //        (0, stream_1.pipeline)(responseStream, requestStream, () => { });
+        //    }
+        //    else {
+        //        requestStream.once('response', () => {
+        //            (0, stream_1.pipeline)(responseStream, requestStream, () => { });
+        //        });
+        //    }
+        // });
+
+
         options.compress = false;
         teenyRequest.stats.requestStarting();
         (0, node_fetch_1.default)(uri, options).then(res => {
-            teenyRequest.stats.requestFinished();
-            responseStream = res.body;
-            responseStream.on('error', (err) => {
-                requestStream.emit('error', err);
-            });
+            teenyRequest.stats.requestFinished(); stream_1.pipeline(res.body, requestStream, () => {});
+        //    responseStream = res.body;
+        //    responseStream.on('error', (err) => {
+        //        requestStream.emit('error', err);
+        //    });
             const response = fetchToRequestResponse(options, res);
             requestStream.emit('response', response);
         }, err => {
             teenyRequest.stats.requestFinished();
             requestStream.emit('error', err);
         });
+
         // fetch doesn't supply the raw HTTP stream, instead it
         // returns a PassThrough piped from the HTTP response
         // stream.
