--- /var/www/sharelatex/web/app/src/router.js
--- /var/www/sharelatex/web/app/src/router.js
@@ -995,11 +995,6 @@ function initialize(webRouter, privateApiRouter, publicApiRouter) {
     AdminController.unregisterServiceWorker
   )

-  privateApiRouter.post(
-    '/disconnectAllUsers',
-    AdminController.disconnectAllUsers
-  )
-
   privateApiRouter.get('/perfTest', (req, res) => res.send('hello'))

   publicApiRouter.get('/status', (req, res) => {
