--- a/settings.coffee
+++ b/settings.coffee
@@ -180,6 +180,11 @@ settings =
 	# cookie with a secure flag (recommended).
 	secureCookie: process.env["SHARELATEX_SECURE_COOKIE"]?
 
+	recaptcha:
+		disabled:
+			invite: true
+			register: true
+
 	# If you are running ShareLaTeX behind a proxy (like Apache, Nginx, etc)
 	# then set this to true to allow it to correctly detect the forwarded IP
 	# address and http/https protocol information.
