--- a/services/web/modules/server-ce-scripts/scripts/check-mongodb.mjs
+++ b/services/web/modules/server-ce-scripts/scripts/check-mongodb.mjs
@@ -6,8 +6,8 @@ import {
 
 const { ObjectId } = mongodb
 
-const MIN_MONGO_VERSION = [6, 0]
-const MIN_MONGO_FEATURE_COMPATIBILITY_VERSION = [6, 0]
+const MIN_MONGO_VERSION = [8, 0]
+const MIN_MONGO_FEATURE_COMPATIBILITY_VERSION = [7, 0]
 
 // Allow ignoring admin check failures via an environment variable
 const OVERRIDE_ENV_VAR_NAME = 'ALLOW_MONGO_ADMIN_CHECK_FAILURES'
